File: killrvideo-commons/src/test/java/com/killrvideo/utils/HashUtilsTest.java
package com.killrvideo.utils;

import org.junit.jupiter.api.Test;

import static com.killrvideo.utils.HashUtils.hashPassword;
import static com.killrvideo.utils.HashUtils.isPasswordValid;
import static org.junit.jupiter.api.Assertions.*;

class HashUtilsTest {

    @Test
    public void testIsPasswordValidWithEmptyRealPasswd() {
        assertFalse(isPasswordValid("", "hash"));
    }

    @Test
    public void testIsPasswordValidWithEmptyHash() {
        assertFalse(isPasswordValid("passwd", ""));
    }

    @Test
    public void testIsPasswordValidWithMismatchPassword() {
        assertFalse(isPasswordValid("passwd", "hash"));
    }

    @Test
    public void testIsPasswordValid() {
        String realPassword = "passwd";
        String hash = hashPassword(realPassword);
        assertTrue(isPasswordValid(realPassword, hash));
    }
}

File: killrvideo-commons/src/test/kotlin/com/killrvideo/utils/HashUtilsTest.kt
package com.killrvideo.utils

import com.killrvideo.utils.HashUtils.hashPassword
import com.killrvideo.utils.HashUtils.isPasswordValid
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

internal class HashUtilsTest {
    @Test
    fun testIsPasswordValidWithEmptyRealPasswd() {
        assertFalse(isPasswordValid("", "hash"))
    }

    @Test
    fun testIsPasswordValidWithEmptyHash() {
        assertFalse(isPasswordValid("passwd", ""))
    }

    @Test
    fun testIsPasswordValidWithMismatchPassword() {
        assertFalse(isPasswordValid("passwd", "hash"))
    }

    @Test
    fun testIsPasswordValid() {
        val realPassword = "passwd"
        val hash: String = hashPassword(realPassword)
        assertTrue(isPasswordValid(realPassword, hash))
    }
}